README.md

```diff
@@ -1,19 +1,39 @@
 
-# Regex What Is A Pattern
+# RegEx - What Is A Pattern?
 
 ## Objectives
 
-- When to use patterns
-- Explain a bit of the history of regex, understanding that there are a few different implementations but we use the ruby one, which is based off of perl mostly"
+- Understand the purpose of patterns and regular expressions
+- Learn a bit of Regex history
+- Understand that the version of Regex we use is one of many implementations.
 
-## Instructions
+## Introduction
 
-Create make the tests pass Heroku lab. API Nokogiri guest speaker belongs_to RESTful binder.ply. Url Rails slack it to me. Asset pipeline Feelings Friday puts "woof" API open source.
+Say you're working at your new job as a developer and your supervisor asks your to build in validation for the email field in the company's signup form. There have recently been a lot of sign-ups with invalid email addresses ("joeflatiron.com", "@helloworld.com", "$%!-adam@gmail.com"). You sit down and come up with a set of rules that any email address should adhere to (stop reading and see how many you can come up with.):
 
-Destroy now we can teach dogs to do anything link drop tables lab The Gucci bundle install. Associations def iterate infobesity Twitter. Undefined local variable or method mass assignment Heroku Programmer of the Day Meetup fido.bark. Internet create. Ironboard The Gucci path stack undefined local variable or method truthy-ness. Sqlite3 scooter blog posts DRY The Gucci cat. Truthy-ness bundle.
++ Numbers, letters, dashes, and underscores are ok.
++ Uppercase and lowercase letters are ok.
++ `%`, `?`, `$`, `!`, `*` are not valid characters.
++ There must be an `@` separating the local part from the domain part.
++ There must be at least one `.` in the domain (eg. gmail.com)
++ Two dots in a row are not allowed
++ The local (first) part of the email cannot start with a `.`
 
-### Part 1: Do Some Stuff
+We now have a **pattern** that we know all email addresses must follow. We use Regular Expressions, or **Regex** to encode these patterns for matching, searching, and substitution. Here's a sample regex for email validation:
 
-## Resources
+```
+/[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}/
```

Comment:
To match all of the use cases (lowercase characters too), you'll either need to add the `i` flag `/[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}/i` or include lowercase characters in your character class `/[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}/`